keps/sig-auth/3926-handling-undecryptable-resources/README.md

```diff
@@ -405,7 +406,115 @@ deletions should therefore have their own extra admission.
 
 The unconditional deletion admission:
 1. checks if a "delete" request contains the `IgnoreStoreReadErrorWithClusterBreakingPotential` option
-2. if it does, it checks the RBAC of the request's user for the `delete-ignore-read-errors` verb of the given resource
+2. if it does, it checks the RBAC of the request's user for the `unsafe-delete-ignore-read-errors` verb of the given resource
+
+
+
+#### Propagating with WATCH
+When a corrupt object is deleted with an `unsafe-delete-ignore-read-errors`, a
+watcher of this resource will not be able to transform its old data from the
+storage, or decode it into an object [1]. This will cause the watch to throw
+an `watch.ERROR` event.
+A client backed up by an informer already has the object in its cache, since the
+client never receives a `watch.DELETED` event the object remains in the lsiter
+cache. This creates an inconsistency - retrieving the object from the cache
+yields the object, but if we get it from the storage we see a `corrupt object`
+error.
+
+There are a few factors we need to consider to understan how the client is impacted:
+- a) a client that has not update yet
+- b) whether watch cache is enabled in the kube-apiserver
+
+We can review the following options:
+
+*A: Send an `ERROR` event with a `metav1.Status` object*:
+When the `watcher` receives a `DELETED` event from etcd, and it fails to
+transform or decode the old data associated with the deleted resource, it
+will return an `ERROR` event with a `metav1.Status` object with a reason
+`StatusReasonStoreReadError`:
+```
+{
+    type: "ERROR",
+    object: {
+      "status": {
+        "message":"corrupt object has been deleted - data from the storage is not transformable - ...",
+		"reason": "StorageReadError"
+	 }
+  }
+}
+```
+
+Upon receiving this error, the existing reflector implementation should end the
+current watch and restart the watch with a new list+watch which will result in
+the old cache be replaced.
+
+- Pros: the existing clients will work without any code change
+- Cons: relisting is expensive
```

Comment:
I think this is acceptable for the situation of a single object being unsafely deleted. Wondering, for the usecase of a lost encryption key, aren't we confronted with potentially many objects? Each will cause a relist 🤔

Reply:
In the whole discussion about downsides of the approaches, wondering about the alternative of doing a manual delete directly through etcd. Where are we that today? How does the apiserver and clients (reflector) behave in that case?

Reply:
I believe manually deleting the etcd key directly will have the same effect, etcd will relay the delete event to any client watching for it.

Reply:
In other words: we don't regress even if we need a client change? That is crucial IMO and should be documented.

Reply:
Rereading your option description: A does not need that client change. And your implementation PR https://github.com/kubernetes/kubernetes/pull/127513 has no client-go changes (beyond the added test).